lambdas/build-stac/utils/events.py

```diff
@@ -15,15 +15,19 @@ class BaseEvent(BaseModel, frozen=True):
     asset_roles: Optional[List[str]] = None
     asset_media_type: Optional[Union[str, pystac.MediaType]] = None
 
+    start_datetime: Optional[constr(regex=r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z?$")] = None
+    end_datetime: Optional[constr(regex=r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z?$")] = None
+    single_datetime: Optional[constr(regex=r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z?$")] = None
```

Comment:
Just a note that you could lean on [Pydantic's to parse the datetime from a string](https://pydantic-docs.helpmanual.io/usage/types/#datetime-types) which would support more flexible parsing (e.g. `2020-01-01T00:00:00+03:00`) or event datetimes without timezones (not sure if this is a good idea or not).  This will also auto-coerce the values to a `datetime` value, meaning you won't need to call `dateutil.parser.parse()` later...

```suggestion
    start_datetime: Optional[datetime] = None
    end_datetime: Optional[datetime] = None
    single_datetime: Optional[datetime] = None
```
Honestly not sure if that is or isn't a good idea though...

Reply:
i believe that should be okay to do!